core/client/fs/src/main/java/alluxio/client/file/AlluxioFileInStream.java

```diff
@@ -302,11 +302,15 @@ private int positionedReadInternal(long pos, byte[] b, int off, int len) throws
       try {
         // Positioned read may be called multiple times for the same block. Caching the in-stream
         // allows us to avoid the block store rpc to open a new stream for each call.
+        BlockInfo blockInfo = lastException == null
+            ? mStatus.getBlockInfo(blockId) : mBlockStore.getInfo(blockId);
```

Comment:
@dbw9580 do you think it's possible that we can use `mStatus.getBlockInfo(blockId)` even if `lastException != null`?